### PR TITLE
Use static closures when not using $this

### PR DESCRIPTION
**Pull request by westonruter**
## What?

Add `static` keyword to closures that don't use `$this`

## Why?

Since closures capture a reference to `$this`, memory leaks can ensue.

This applies the same change that was just made in Core-58323

## How?

I used the `SlevomatCodingStandard.Functions.StaticClosure` sniff to apply the change with `phpcbf`.

## Testing Instructions

n/a

### Testing Instructions for Keyboard

n/a

### CODE REVIEW

**Comment by gziolo** [comment]
I see that @aristath opened another PR https://github.com/WordPress/gutenberg/pull/50657 with similar changes. 

CI errors reported are unrelated to this PR. I belive it must be reported on `trunk`, too.

**Comment by aristath** [comment]
> I see that @aristath opened another PR https://github.com/WordPress/gutenberg/pull/50657 with similar changes.

Yes, it was almost identical, inspired by @westonruter's PR in https://github.com/WordPress/performance/pull/729 👍 
I closed the other PR, we can continue with this one 🎉

File: packages/block-library/src/comments-pagination-next/index.php
Comment (spacedmonkey) [inline packages/block-library/src/comments-pagination-next/index.php:32]:
```suggestion
	add_filter( 'next_comments_link_attributes', 'get_block_wrapper_attributes' );
```

Why not just do this?

File: packages/block-library/src/comments-pagination-previous/index.php
Comment (spacedmonkey) [inline packages/block-library/src/comments-pagination-previous/index.php:28]: See https://github.com/WordPress/gutenberg/pull/50723/files#r1198736035

**Review by spacedmonkey — COMMENTED** [review]
This is nearly there. I have one nitpic, then I will approve.

File: packages/block-library/src/comments-pagination-next/index.php
Comment (spacedmonkey) [reply packages/block-library/src/comments-pagination-next/index.php:32]: I think I know why this is done. `next_comments_link_attributes` the value is a string and `get_block_wrapper_attributes` expects an array.

**Comment by gziolo** [comment]
> This is nearly there. I have one nitpic, then I will approve.

Once approved, it can be merged with the admin permissions if the JS linting errors are the only blocker in the CI check.

File: packages/block-library/src/comments-pagination-next/index.php
Comment (westonruter) [reply packages/block-library/src/comments-pagination-next/index.php:32]: Yes, I see that `get_block_wrapper_attributes()` takes an `$extra_attributes` array arg.

File: packages/block-library/src/comments-pagination-previous/index.php
Comment (westonruter) [reply packages/block-library/src/comments-pagination-previous/index.php:28]: Resolving per above.

**Review by spacedmonkey — APPROVED** [review]

**Comment by westonruter** [comment]
@aristath Sorry that I didn't check for your PR first!

What do we think about including `slevomat/coding-standard` in the [`phpcs.xml.dist`](https://github.com/WordPress/gutenberg/blob/trunk/phpcs.xml.dist)? It was discouraged from core in https://github.com/WordPress/wordpress-develop/pull/4457 for Core-58323.

**Comment by westonruter** [comment]
So include `slevomat/coding-standard` or not?

**Review by aristath — APPROVED** [review]
This looks good to me 👍 

> So include `slevomat/coding-standard` or not?

I suppose we could merge the improvements from this PR without including `slevomat/coding-standard` just to get this part out of the way, and create a separate issue to include the `slevomat/coding-standard` package where we can focus more on that discussion 👍

**Pull request merged**
